src/include/http.h

```diff
@@ -53,10 +54,53 @@
      (record->app == 80 || (record->key.sp == 80 || record->key.dp == 80)) \
     )
 
+enum http_line_type {
+    HTTP_LINE_RESERVED  = 0,
+    HTTP_LINE_REQUEST   = 1,
+    HTTP_LINE_STATUS    = 2,
+};
+
+struct http_header_status_line {
+    char *version;
+    char *code;
+    char *reason;
+};
+
+struct http_header_request_line {
+    char *method;
+    char *uri;
+    char *version;
+};
+
+struct http_header_element {
+    char *name;
+    char *value;
+};
+
+#define HTTP_MAX_HEADER_ELEMENTS 32
+
+struct http_header {
```

Comment:
Just a general question, why do you not use typedefs for structures?

Reply:
That's just the style that most of the code base has used since beginning.
Personally, keeping struct makes it more visually distinct for me when reading.